redbeams/src/main/java/com/sequenceiq/redbeams/service/network/SubnetChooserService.java

```diff
@@ -0,0 +1,30 @@
+package com.sequenceiq.redbeams.service.network;
+
+import java.util.List;
+
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.cloud.model.CloudSubnet;
+import com.sequenceiq.cloudbreak.exception.BadRequestException;
+
+@Service
+public class SubnetChooserService {
+
+    public List<CloudSubnet> chooseSubnetsFromDifferentAzs(List<CloudSubnet> subnetMetas) {
+        if (subnetMetas.size() < 2) {
+            throw new BadRequestException("Insufficient number of subnets: at least two subnets in two different availability zones needed");
```

Comment:
I think RedbeamsException is better here, because this condition is not because there was something (directly) wrong with the API request. This isn't a blocker for me, though. Your call.

Reply:
Ok, I did change this to redbeams exception. This will be mapped to INTERNAL_SERVER_ERROR. To the end user this perfectly makes sense. On the other hand, to the direct consumers of our api I am not sure if this is ok.